src/vs/editor/contrib/stickyScroll/browser/stickyScrollProvider.ts

```diff
@@ -96,9 +97,28 @@ export class StickyLineCandidateProvider extends Disposable {
 		}
 	}
 
+	private updateIndex(index: number) {
+		if (index === -1) {
+			index = 0;
+		} else if (index < 0) {
+			index = -index - 2;
+		}
+		return index;
+	}
+
 	public getCandidateStickyLinesIntersectingFromOutline(range: StickyRange, outlineModel: StickyOutlineElement, result: StickyLineCandidate[], depth: number, lastStartLineNumber: number): void {
+		if (outlineModel.children.length === 0) {
+			return;
+		}
 		let lastLine = lastStartLineNumber;
-		for (const child of outlineModel.children) {
+		const childrenStartLines = outlineModel.children.map(child => child.range?.startLineNumber as number);
```

Comment:
The `childrenStartLines`-array can contain `undefined` and numbers. That is likely unexpected and maybe something that you should check for. Like, ignore those element that have no range